probe-rs/src/debug/unit_info.rs

```diff
@@ -142,43 +137,48 @@ impl UnitInfo {
+            let mut die_ranges = Vec::new();
+            while let Ok(Some(gimli_range)) = gimli_ranges.next() {
+                die_ranges.push(gimli_range.begin..gimli_range.end);
+            }
 
-            while let Ok(Some(range)) = ranges.next() {
-                if !range.contains(address) {
-                    continue;
-                }
-                // Check if we are actually in an inlined function
+            if !die_ranges.iter().any(|range| range.contains(&address)) {
+                continue;
+            }
 
-                // We don't have to search further up in the tree, if there are multiple inlined functions,
-                // they will be children of the current function.
-                abort_depth = current_depth;
+            // Check if we are actually in an inlined function
 
-                // Find the abstract definition
-                let Ok(Some(abstract_origin)) = current.attr(gimli::DW_AT_abstract_origin) else {
-                    tracing::warn!("No abstract origin for inlined function, skipping.");
-                    return Ok(vec![]);
-                };
-                let abstract_origin_value = abstract_origin.value();
-                let gimli::AttributeValue::UnitRef(unit_ref) = abstract_origin_value else {
-                    tracing::warn!(
-                        "Unsupported DW_AT_abstract_origin value: {:?}",
-                        abstract_origin_value
-                    );
-                    continue;
-                };
+            // We don't have to search further up in the tree, if there are multiple inlined functions,
+            // they will be children of the current function.
+            abort_depth = current_depth;
 
-                let Some(mut die) = self.unit.entry(unit_ref).ok().and_then(|abstract_die| {
-                    FunctionDie::new_inlined(current.clone(), abstract_die.clone(), self)
-                }) else {
-                    continue;
-                };
+            // Find the abstract definition
+            let Ok(Some(abstract_origin)) = current.attr(gimli::DW_AT_abstract_origin) else {
+                tracing::warn!("No abstract origin for inlined function, skipping.");
+                return Ok(vec![]);
+            };
+            let abstract_origin_value = abstract_origin.value();
+            let gimli::AttributeValue::UnitRef(unit_ref) = abstract_origin_value else {
+                tracing::warn!(
+                    "Unsupported DW_AT_abstract_origin value: {:?}",
+                    abstract_origin_value
+                );
+                continue;
+            };
 
-                die.low_pc = range.begin;
-                die.high_pc = range.end;
+            let Some(die) = self.unit.entry(unit_ref).ok().and_then(|abstract_die| {
+                FunctionDie::new_inlined(current.clone(), abstract_die.clone(), self).map(
+                    |mut inlined_function_die: FunctionDie<'_, '_, '_>| {
```

Comment:
Do we need the type hint here? If this is just for readability, I think it's obvious enough that `FunctionDie::new_inlined` returns something that contains a `FunctionDie`. If we need the type hint, do we need to list the elided lifetimes?